dotnet/src/SemanticKernel/Planning/ConditionalFlowHelper.cs

```diff
@@ -0,0 +1,367 @@
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Text.Json;
+using System.Text.Json.Nodes;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+using System.Xml;
+using Microsoft.Extensions.Logging;
+using Microsoft.SemanticKernel.AI.TextCompletion;
+using Microsoft.SemanticKernel.Orchestration;
+
+namespace Microsoft.SemanticKernel.Planning;
+
+/// <summary>
+/// <para>Semantic skill that evaluates conditional structures</para>
+/// <para>
+/// Usage:
+/// var kernel = SemanticKernel.Build(ConsoleLogger.Log);
+/// kernel.ImportSkill("conditional", new ConditionalSkill(kernel));
+/// </para>
+/// </summary>
+public class ConditionalFlowHelper
+{
+    private readonly IKernel _kernel;
+
+    #region Prompts
+
+    internal const string IfStructureCheckPrompt =
+        @"Structure:
+<if condition="""">
+</if>
+<else>
+</else>
+
+Rules:
+. A condition attribute must exists in the if tag
+. ""If"" tag must have one or more children nodes
+. ""Else"" is optional
+. If ""Else"" is provided must have one or more children nodes
+. Return true if Test Structure is valid
+. Return false if Test Structure is not valid with a reason with everything that is wrong
+. Give a json list of variables used inside the attribute ""condition"" of the first ""IF"" only
+. All the return should be in Json format.
+. Response Json Structure:
+{
+    ""valid"": bool,
+    ""reason"": string,
+    ""variables"": [string] (only the variables within ""Condition"" attribute)
+}
+
+Test Structure:
+{{$IfStatementContent}}
+
+Response: ";
+
+    internal const string EvaluateConditionPrompt =
+        @"Rules
+. Response using the following json structure:
+{ ""valid"": bool, ""condition"": bool, ""reason"": string }
+
+. A list of variables will be provided for the condition evaluation
+. If condition has an error, valid will be false and property ""reason"" will have all detail why.
+. If condition is valid, update reason with the current evaluation detail.
+. Return ""condition"" as true or false depending on the condition evaluation
+
+Variables Example:
+x = 2
+y = ""adfsdasfgsasdddsf""
+z = 100
+w = ""sdf""
+
+Given Example:
+$x equals 1 and $y contains 'asd' or not ($z greaterthan 10)
+
+Reason Example:
+(x == 1 ∧ (y contains ""asd"") ∨ ¬(z > 10))
+(TRUE ∧ TRUE ∨ ¬ (FALSE))
+(TRUE ∧ TRUE ∨ TRUE)
+TRUE
+
+Variables Example:
+24hour = 11
+
+Given Example:
+$24hour equals 1 and $24hour greaterthan 10
+
+Response Example:
+{ ""valid"": true, ""condition"": false, ""reason"": ""(24hour == 1 ∧ 24hour > 10) = (FALSE ∧ TRUE) = FALSE"" }
+
+Variables Example:
+24hour = 11
+
+Given Example:
+Some condition
+
+Response Example:
+{ ""valid"": false, ""reason"": ""<detail why>"" }
+
+Variables Example:
+a = 1
+b = undefined
+c = ""dome""
+
+Given Example:
+(a is not undefined) and a greaterthan 100 and a greaterthan 10 or (a equals 1 and a equals 10) or (b is undefined and c is not undefined)
```

Comment:
This is far more complex than we need

Reply:
Yep, we might get this extra bonus from the LLM evaluation. I will be leaving this open for now, if this gives problems we can enforce what are the allowed comparison operators.

I will enforce that this PR we will support `equals` and more simple operations first (greater than, less than) and may change if any `LLM >= GPT4` get it off rails.